loopy/target/pyopencl.py

```diff
@@ -799,7 +799,9 @@ def get_kernel_call(self, codegen_state, name, gsize, lsize, extra_args):
             Comment("{{{ enqueue %s" % name),
             Line(),
             Assign("_lpy_knl", "_lpy_cl_kernels."+name),
-            Assert("_lpy_knl.num_args == %d" % cl_arg_count),
+            Assert(f"_lpy_knl.num_args == {cl_arg_count}, f'Kernel \"{name}\" "
+                   f"setup arg count ({cl_arg_count}) does not match actual arg "
+                   "count ({_lpy_knl.num_args}).'"),
```

Comment:
```suggestion
                   f"count ({_lpy_knl.num_args}).'"),
```

Reply:
This suggestion isn't correct I think. The generated code needs to evaluate `{_lpy_knl.num_args}` at runtime, not here where it is generated.

Reply:
Ah! You're right. Add a comment, otherwise I bet someone else will try to "fix" this. :)